### PR TITLE
Sprint 13 (dev->staging)

### PR DESCRIPTION
**Pull request by afrasier**
This PR includes:

Frontend updates:
- PR #310 
- PR #311 
- PR #313 

As well as:
- #314 Task model, endpoints
- #315 Historical views
- #316 Prevent bids where bidder will retire

### CODE REVIEW

File: talentmap_api/common/common_helpers.py [hunk 1]
```diff
@@ -285,3 +287,11 @@ def xml_etree_to_dict(tree):
     dictionary[tree.tag] = tree.text
 
     return dictionary
+
+
+def serialize_instance(instance, serializer_string, many=False):
+    '''
+    Used when performing some look-up logic in a serializer
+    Returns the objects serialized data.
```
Comment (jseppi) [inline talentmap_api/common/common_helpers.py]: `object's`

File: talentmap_api/bidding/views/bidlist.py [hunk 2]
```diff
@@ -100,7 +102,15 @@ def put(self, request, pk, format=None):
         bid = Bid.objects.get_or_create(bidcycle=bidcycle,
                                         user=UserProfile.objects.get(user=self.request.user),
                                         position=position)[0]
-        UserProfile.objects.get(user=self.request.user).bidlist.add(bid)
+        user = UserProfile.objects.get(user=self.request.user)
+
+        # User cannot be retiring during the positions tour of duty
```
Comment (jseppi) [inline talentmap_api/bidding/views/bidlist.py]: `position's`

File: talentmap_api/bidding/urls/bidcycle.py [hunk 3]
```diff
@@ -1,8 +1,12 @@
 from django.conf.urls import url
+from rest_framework import routers
 
 from talentmap_api.bidding.views import bidcycle as views
 from talentmap_api.common.urls import get_retrieve, patch_update, post_create, get_list
 
+router = routers.SimpleRouter()
+router.register(r'^(?P<instance_id>[0-9]+)/history', views.HistoricalBidCycleView, base_name="organization.BidCycle-history")
```
Comment (jseppi) [inline talentmap_api/bidding/urls/bidcycle.py]: should this be `base_name="bidding.BidCycle-history"`?

File: talentmap_api/common/history_helpers.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                "history_user": FOREIGN_KEY_LOOKUPS
+            }
+
+    class HistoricalSerializer(serializer):
+        history_user = StaticRepresentationField(read_only=True)
+
+        class Meta(serializer.Meta):
+            model = model_class.history.model
+            fields = "__all__"
+            nested = {}  # No nesting serializers here - the FK traversal doesn't cascade (yet)
```
Comment (jseppi) [inline talentmap_api/common/history_helpers.py]: 🔜

File: talentmap_api/common/tests/test_historical_views.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+from model_mommy import mommy
+
+
+@pytest.mark.django_db()
+def test_post_historical_view(client):
+    with freeze_time("1990-01-01T00:00:00Z") as frozen_time:
+        tod = mommy.make('organization.TourOfDuty', long_description="2YR", months=24)
+        post = mommy.make('organization.Post', danger_pay=10, differential_rate=10, tour_of_duty=tod)
+
+        frozen_time.move_to("1995-01-01T00:00:00Z")
```
Comment (jseppi) [inline talentmap_api/common/tests/test_historical_views.py]: ❄️

File: talentmap_api/common/history_helpers.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+from talentmap_api.user_profile.models import UserProfile
+from talentmap_api.user_profile.filters import UserProfileFilter
+
+from talentmap_api.common.serializers import StaticRepresentationField
+from talentmap_api.common.filters import INTEGER_LOOKUPS, ALL_TEXT_LOOKUPS, DATE_LOOKUPS, FOREIGN_KEY_LOOKUPS
+from talentmap_api.common.mixins import FieldLimitableSerializerMixin
+
+
+def generate_historical_view(model_class, serializer, filter):
```
Comment (jseppi) [inline talentmap_api/common/history_helpers.py]: 👍

File: talentmap_api/bidding/urls/bidcycle.py (hunk 3 above)
Comment (afrasier) [reply talentmap_api/bidding/urls/bidcycle.py]: Ah, yes. It's not actually used anywhere at the moment fortunately. Will update.

**Pull request merged**